File: musicShopProject/tentashop/src/domain/address/addressBlank.ts
import { Address } from "./address";

export interface AddressBlank{
         id: string | null,
         city: string | null,
         street: string | null,
         home: string | null,
         apartment: string | null,
}

export namespace AddressBlank {
    export function getEmpty(addresId: string): AddressBlank {
        return {
             id: addresId,
         city: null,
         street: null,
         home: null,
         apartment: null,
        };
    }

    export function toProductBlank(address: Address): AddressBlank {
        return {
            id: address.id,
         city: address.city,
         street: address.street,
         home: address.home,
         apartment: address.apartment,
        };
    }
 }

File: musicShopProject/Model/Addresses/AddressBlank.cs
namespace musicShopProject.Model.Addresses;

public partial class AddressBlank
{
    public Guid? Id { get; set; }
    public String? City { get; set; }
    public String? Street { get; set; }
    public String? Home { get; set; }
    public String? Apartment { get; set; }
}

public partial class AddressBlank
{
    public class Validated
    {
        public Guid Id { get; }
        public String City { get;  }
        public String Street { get; }
        public String Home { get; }
        public String Apartment { get; }

        public Validated(Guid id, String city, String street, String home, String apartment)
        {
            Id = id;
            City = city;
            Street = street;
            Home = home;
            Apartment = apartment;
        }
    }
}
